### PR TITLE
[Editorial] Max points on a line

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/platinum/leetcode-149.mdx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+author: Mihnea Brebenel
+---
+
+## Explanation
+
+Because $n \le 300$ we can use a brute-force approach: choose two points and check if other point are collinear with them.
+Three points $A$, $B$, and $c$ are collinear if and only if $AB$ has the same slope as $BC$:
+
+$$
+\frac{A.y-B.y}{A.x-B.x}=\frac{B.y-C.y}{B.x-C.x}
```
Comment (SansPapyrus683) [inline solutions/platinum/leetcode-149.mdx:14]: perhaps use $a_x$, $b_y$, etc.

File: solutions/platinum/leetcode-149.mdx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+		int ans = 0;
+		for (int i = 0; i < points.size(); i++) {
+			for (int j = i + 1; j < points.size(); j++) {
+				int p = 0;
+				for (int k = j + 1; k < points.size(); k++) {
+					if ((points[j][1] - points[i][1]) *
+					        (points[i][0] - points[k][0]) ==
+					    (points[i][1] - points[k][1]) *
+					        (points[j][0] - points[i][0])) {
```
Comment (SansPapyrus683) [inline solutions/platinum/leetcode-149.mdx:40]: put the expressions in this if statement into vars like `lhs` and `rhs` and add a comment "check if [smth] per the formula given above"

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
might just be me, but your code does seem to be getting better

File: solutions/platinum/leetcode-149.mdx (hunk 1 above)
Comment (brebenelmihnea) [reply solutions/platinum/leetcode-149.mdx]: done

File: solutions/platinum/leetcode-149.mdx (hunk 2 above)
Comment (brebenelmihnea) [reply solutions/platinum/leetcode-149.mdx]: done

**Comment by brebenelmihnea** [comment]
https://leetcode.com/problems/max-points-on-a-line/submissions/1305898431/
image

**Review by SansPapyrus683 — APPROVED** [review]
bro why'd you undo my code changes 💀

**Review by ryanchou-dev — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
there's merge conflicts 💀

**Comment by brebenelmihnea** [comment]
it was a conflict with my other PR 💀

**Pull request merged**